app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt

```diff
@@ -57,7 +64,19 @@ class SearchFragment : Fragment() {
         }
 
         val time = safeArgs?.stringSavedDate
-        rootView.timeTextView.text = time ?: getString(R.string.anytime)
+        if (time.isNullOrBlank()) rootView.timeTextView.text = getString(R.string.anytime)
+        else {
+            try {
+                val zonedDate =
+                    LocalDate.parse(time, DateTimeFormatter.ofPattern("yyyy-MM-dd"))
+                        .atStartOfDay(ZoneId.systemDefault()).plusDays(-1)
+                if (zonedDate.year == Calendar.getInstance().get(Calendar.YEAR))
+                    rootView.timeTextView.text = getFormattedDateWithoutYear(zonedDate)
+                else rootView.timeTextView.text = getFormattedDate(zonedDate)
+            } catch (e: DateTimeParseException) {
+                rootView.timeTextView.text = time
```

Comment:
Here time is null or blank, right?

Reply:
No, I checked that before. Here time is today, tomorrow, weekend, rtc